starfish/image/_stack.py

```diff
@@ -45,6 +53,107 @@ def numpy_array(self, data):
         self._data_needs_writeback = True
         data.setflags(write=False)
 
+    def get_slice(
+            self,
+            indices: typing.Mapping[Indices, typing.Union[int, slice]]
+    ) -> typing.Tuple[numpy.ndarray, typing.Sequence[Indices]]:
+        """
+        Given a dictionary mapping the index name to either a value or a slice range, return a numpy array representing
+        the slice, and a list of the remaining axes beyond the normal x-y tile.
+
+        Example:
+            ImageStack axes: H, C, and Z with shape 3, 4, 5, respectively.
+            ImageStack Implicit axes: X, Y with shape 10, 20, respectively.
+            Called to slice with indices {Z: 5}.
+            Result: a 4-dimensional numpy array with shape (3, 4, 10, 20) and the remaining axes [H, C].
+
+        Example:
+            Original axes: H, C, and Z.
+            Implicit axes: X, Y.
+            Called to slice with indices {Z: 5, C: slice(2, 4)}.
+            Result: a 4-dimensional numpy array with shape (3, 2, 10, 20) and the remaining axes [H, C].
```

Comment:
Same comment as above, but otherwise I think these two examples make really clear what you can do here. 👍